### PR TITLE
Add tests on snapshots of graphical output

### LINKED ISSUE #13: Update tests
Just to note that snapshots of graphical output can be tested using [`vdiffr`](https://vdiffr.r-lib.org/) package

_Originally posted by @annakrystalli in https://github.com/Infectious-Disease-Modeling-Hubs/hubVis/pull/2#discussion_r1374730843_

### PR DESCRIPTION
**Pull request by LucieContamin**
Fixes #13

### CODE REVIEW

**Review by zkamvar — CHANGES_REQUESTED** [review]
This looks good! 

My suggestions are:

1. remove the `library()` calls from the test files. They cause pollution of the R environment.
2. When you test for errors and warnings, add a message to make sure that the error/warning the test is producing is actually the one you are testing for: https://r-pkgs.org/testing-basics.html#testing-errors. I would be happy to help with this if you would like. 
3. A lot of the tests cause plots to be created and displayed in my browser. If you do something like:
   ```diff
   - expect_warning(plot_step_ahead_model_output(df_test, target_data_us))
   + expect_warning(res <- plot_step_ahead_model_output(df_test, target_data_us))
   ```
   then the plot will not print.

File: tests/testthat/test-plot_step_ahead_model_output.R [hunk 1]
```diff
@@ -1,56 +1,72 @@
 # Data Preparation
 library(hubExamples)
+library(vdiffr)
```
[the remark this answers is not in the document]
Comment (LucieContamin) [reply tests/testthat/test-plot_step_ahead_model_output.R:3]: Good catch! thank you! I forgot to remove it.

**Review by zkamvar — APPROVED** [review]

File: tests/testthat/test-palette_utils.R [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# Test input information
+test_that("Palette Input parameters", {
+
+  ## Palette/Color input
+  ### One color
+  expect_snapshot(plot_step_ahead_model_output(projection_data_a_us,
+                                               target_data_us,
+                                               show_plot = FALSE,
+                                               pal_color = NULL,
+                                               one_color = "set2"))
```
Comment (zkamvar) [inline tests/testthat/test-palette_utils.R:28]: The snapshot tests are great if you want to move fast, but this is not a great indicator of what the test is actually doing. 

Instead, I would suggest to use a part of the actual warning message to test that it's producing the correct warning. 

Wrapping it in `suppressWarnings()` prevents R from complaining about warnings during tests. 

```suggestion
  suppressWarnings({
    expect_warning(plot_step_ahead_model_output(projection_data_a_us,
                                                 target_data_us,
                                                 show_plot = FALSE,
                                                 pal_color = NULL,
                                                 one_color = "set2"),
      "is not one of the accepted color name"
    )
  })
```

**Review by zkamvar — CHANGES_REQUESTED** [review]
I'm glad you split some of the tests out into separate files, but I would suggest to use `expect_warning()` and `expect_error()` instead of `expect_snapshot()` because snapshot tests can be quite  brittle and break unexpectedly due to random formatting changes.

File: tests/testthat/test-palette_utils.R (hunk 2 above)
Comment (LucieContamin) [reply tests/testthat/test-palette_utils.R]: Done! Thank you for the feedback!!

**Review by zkamvar — APPROVED** [review]
Looks good! Thank you for your patience with the test feedback.

**Pull request merged**